dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -985,4 +1045,108 @@ private void addLocationHeader(HttpHeaders headers, PersistentEntityResourceAsse
         repository.delete(id);
         return ControllerUtils.toEmptyResponse(HttpStatus.NO_CONTENT);
     }
+
+    /**
+     * Execute a PUT request for an entity with id of type Integer;
+     *
+     * curl -X PUT -H "Content-Type:application/json" http://<dspace.url>/dspace-spring-rest/api/{apiCategory}/{model}/{id}
+     *
+     * Example:
+     * <pre>
+     * {@code
+     *      curl -X PUT -H "Content-Type:application/json" http://<dspace.url>/dspace-spring-rest/api/core/metadatafield/1
+     * }
+     * </pre>
+     *
+     * @param request     the http request
+     * @param apiCategory the API category e.g. "api"
+     * @param model       the DSpace model e.g. "collection"
+     * @param id        the ID of the target REST object
+     * @param jsonNode    the part of the request body representing the updated rest object
+     * @return the relevant REST resource
+     */
+    @RequestMapping(method = RequestMethod.PUT, value = REGEX_REQUESTMAPPING_IDENTIFIER_AS_DIGIT,
+        consumes = {"application/json", "application/hal+json"})
+    public DSpaceResource<RestAddressableModel> put(HttpServletRequest request,
```

Comment:
Same here (as POST).  We now seem to have two `PUT` options, when I think we should only be supporting "text/uri-list" (per our previous discussions in this PR).

Reply:
The explanation here is similar to 
https://github.com/DSpace/DSpace/pull/2332#discussion_r256743934
The PUT method with JSON is still required for other classes. If you'd use `PUT` with JSON for the `RelationshipRestRepository`, it will throw a `RepositoryMethodNotImplementedException`

Reply:
Marking this conversation as resolved. See other comment/explanation. I misunderstood the purpose of these methods. Thanks for clarifying